### PR TITLE
Resolves #51 add string limits to create proposal

### LINKED ISSUE #51: Create proposal will fail without an error if string contents are too long
No checks are made on the length of strings in the create proposal form.  Needs to behave more like updating a proposal summary.

### PR DESCRIPTION
**Pull request by AllanEngland**
And a character counter from the the edit summary pane, and apply limit to new title and edit title.

### CODE REVIEW

File: src/main/webui/src/proposal/New.tsx [hunk 1]
```diff
@@ -62,8 +63,23 @@ const kindData = [{value: "STANDARD", label: "Standard"}, {value: "TOO", label:
             }
             <form onSubmit={createNewObservingProposal}>
                 <Box>
-                <TextInput name="title" placeholder="Give your proposal a title" withAsterisk label={"Title"} {...form.getInputProps("title")}/>
-                <Textarea rows={3} name="summary" placeholder="A brief summary" withAsterisk label={"Summary"} {...form.getInputProps("summary")} />
+                <TextInput name="title"
+                           maxLength={MAX_CHARS_FOR_INPUTS}
+                           placeholder="Give your proposal a title"
+                           withAsterisk
+                           label={"Title"}
```
Comment (alan-stokes) [inline src/main/webui/src/proposal/New.tsx:70]: should we not now also have a "char remaining for this field as well now?

File: src/main/webui/src/proposal/Title.tsx [hunk 2]
```diff
@@ -74,7 +75,9 @@ function TitlePanel() {
             { isLoading ? ("Loading..") :
                  submitting ? ("Submitting..."):
             <form onSubmit={updateTitle}>
-                <TextInput name="title" {...form.getInputProps('title')}/>
+                <TextInput name="title"
+                           maxLength={MAX_CHARS_FOR_INPUTS}
+                           {...form.getInputProps('title')}/>
```
Comment (alan-stokes) [inline src/main/webui/src/proposal/Title.tsx:80]: same question. chars remaining component?

**Review by alan-stokes — CHANGES_REQUESTED** [review]

**Comment by alan-stokes** [comment]
the fact we're talking about putting this stuff in many places now. makes me wonder if we should have a component / function to fold this in by default? just needing the form name as a input afterall

File: src/main/webui/src/proposal/New.tsx (hunk 1 above)
Comment (AllanEngland) [reply src/main/webui/src/proposal/New.tsx:70]: I thought about this when I did it, in my opinion I didn't think it necessary and would create clutter.  However, I'm no expert in this field and I've never written an observing proposal, so is 255 chars potentially restricting titles?

**Comment by AllanEngland** [comment]
I tried a few different approaches and ended up going with the simplest.  A common characters remaining element has been added to the appropriate fields.

File: src/main/webui/src/commonInputs/remainingCharacterCount.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+/**
+ * used by the MaxCharsForInputRemaining function.
+ *
+ * @param {number} length: the current length of the string.
+ */
+interface CharCountInputProps {
+    length: number
+    //isDimmed?: boolean
```
Comment (alan-stokes) [inline src/main/webui/src/commonInputs/remainingCharacterCount.tsx:13]: if this is commented out, why keep it?

**Comment by alan-stokes** [comment]
looks fine (once that 1 liner is removed [assuming thats the case]) do have a question for beyond this pr. which is, why is it that the summary wraps around when it hits the end, but the title keeps scrolling? is this something we'd want to make consistent that all our input boxes wrap?

**Comment by AllanEngland** [comment]
> looks fine (once that 1 liner is removed [assuming thats the case]) do have a question for beyond this pr. which is, why is it that the summary wraps around when it hits the end, but the title keeps scrolling? is this something we'd want to make consistent that all our input boxes wrap?

Surely anything beyond this pr should be a new discussion or issue?

**Pull request merged**